### ISSUE TITLE
Prompt file type after creating a new file

### ISSUE THREAD

**Issue by digitarald** [microsoft/vscode#111053]
tl;dr: Make is easier to start coding in new files by prompting users to set the language mode earlier.

### Problems

Creating new files results in a plain text file until the file is saved or a language mode is set. In researching new users, this confused both people new to coding and experienced IDE users.

Saving a file without a language mode set also causes the save dialog to not have a default file extension (in windows the extension list can be long and unwieldy).

#### Prompt for language mode after New File opens

One possible solution is that "New File" open the language mode picker right after opening the blank file. The prompt can still be closed if a plain text file is what the user wants.

A risk that was called out is that users might expect the first prompt to name the file. The language prompt picker does not allow any input, so this wouldn't be accepted. As the only label for the picker is the input placeholder, UX could be improved to have clearer instructions for the picker even after users start typing.

#### Prompt for language mode before saving a new file

Alternatively, the language prompt can show before saving to make the save dialog easier to use. This means until saving most user will not have any language features.

#### Nudge to set language mode

As a onboarding bandaid, the UI could annotate/highlight the language element in the status bar; but now users tend to miss those subtle nudges.

**Comment by sana-ajani** [comment]
Related to this: https://github.com/microsoft/vscode/issues/110330

I think we could run this one as an A/B test. The current implementation that Christof added prompts users with the save dialog when clicking "New File". The test could be prompting users with the language mode picker and we can measure if users are more likely to install the recommended extension and other actions from the newly created file.

**Comment by chrisdias** [comment]
related, I wonder if we could do a watermark, similar to repl.it which is using the Monaco editor.

![image](https://user-images.githubusercontent.com/1487073/99850838-f133e000-2b32-11eb-981b-b62109ea7464.png)

**Comment by misolori** [comment]
I think this has always been modeled after current text editors like Sublime and Textmate. If the behavior were to ever be changed, we'd definitely need a setting to toggle it. Changing it does add friction when you want to quickly write down notes and could annoy current users. Do we have any open issues related to this?

**Comment by digitarald** [comment]
> I think this has always been modeled after current text editors like Sublime and Textmate.

That origin makes sense. One counter argument to this is that VS Code diverged from those basic code editors with code intelligence as a core value proposition (which doesn't work in a plain text file).

> Changing it does add friction when you want to quickly write down notes and could annoy current users.

Agreed on having a pref for it (advanced users already have `files.defaultLanguage`, maybe this can build on it). I'd hypothesize that the risk of frustration for power users is small as the picker can be quickly dismissed by `Esc` and click.

**Comment by sana-ajani** [comment]
I'd also suggest only having this language mode choice when a user clicks on"new file" from the welcome page. Hypothesis here is that advanced users likely create new files in existing or open folders, or using` cmd+n` and we don't want to add that step there.
